app/assets/javascripts/host_edit.js

```diff
@@ -295,10 +295,7 @@ function serializeForm() {
 
 //This function decides for a given input element is it set explicitly by the user.
 function isExplicit(index, element) {
-  if (element.nextSibling == undefined) return true;
-  if (element.nextSibling.children == undefined) return true;
-  if (element.nextSibling.children[0] == undefined) return true;
-  if (element.nextSibling.children[0].dataset == undefined) return true;
+  if (!element.classList.contains("has-explicit")) return true;
   return element.nextSibling.children[0].dataset.explicit == 'true';
```

Comment:
Perhaps solve two comments with one step - by putting the class on the button rather than the select element, the `return` condition could check if the button element exists with the right class, *and* this line could reference the same button element.